Resources/public/js/models/ez-locationmodel.js

```diff
@@ -22,6 +22,34 @@ YUI.add('ez-locationmodel', function (Y) {
      */
     Y.eZ.Location = Y.Base.create('locationModel', Y.eZ.RestModel, [], {
         /**
+         * Override of the eZ.RestModel _parseStruct method to also read the content info
+         *
+         * @protected
+         * @method _parseStruct
+         * @param {Object} struct the struct to transform
+         * @return {Object}
+         */
+        _parseStruct: function (struct) {
+            var attrs;
+
+            attrs = this.constructor.superclass._parseStruct.call(this, struct);
+            attrs.contentInfo = {
```

Comment:
this seems very complicated and actually you are parsing the ContentInfo structure in a Location why not just:
```js
attrs.contentInfo = struct.ContentInfo;
```

the ContentInfo has rules to pick the necessary properties and ignore the rest (that's the point of the model ;))